core/domain/param_domain.py

```diff
@@ -26,19 +26,47 @@
 
 class ParamSpec(object):
     """Value object for an exploration parameter specification."""
+
     def __init__(self, obj_type):
+        """Initializes a ParamSpec object with the specified object type.
+
+        Args:
+            obj_type: unicode. The object type with which the parameter is
+                initialized.
+        """
         self.obj_type = obj_type
 
+
     def to_dict(self):
+        """Returns a dict representation of this ParamSpec.
+
+        Returns:
+            dict. A dict with a single key, whose value is the type
+                of the parameter represented by this ParamSpec.
+        """
         return {
             'obj_type': self.obj_type,
         }
 
+
     @classmethod
     def from_dict(cls, param_spec_dict):
+        """Creates a ParamSpec object from its dict representation.
+        Args:
```

Comment:
Here and elsewhere: there should be a single newline above `Args:`. Follow the format of existing docstrings.